### ISSUE TITLE
How Juseppe is used to upgrade and downgrade plugins?  How Juseppe handles or manages dependencies?

### ISSUE THREAD

**Issue by itshemant4u** [yandex-qatools/juseppe#14]
Please help me understand how Juseppe is used to upgrade and downgrade plugins. Also, how Juseppe handles or manages dependencies?

**Comment by lanwen** [comment]
Juseppe itself don't handle anything - it just exposes all plugin files with all required info.
Juseppe exposes latest uploaded version plugin if there is a couple of version of one plugin

**Comment by itshemant4u** [comment]
Thanks for your response. I want to know how Juseppe is used to install plugins. I am manually uploading plugins under /var/lib/juseppe/plugins 
I have also configured 'Manage UpdateSites' plugin. 

**However, the plugins from /var/lib/juseppe/plugins are not uploading on my Jenkins server. Please let me know what might be missing? **

**Comment by itshemant4u** [comment]
Hi Kirill,

I want to request you to respond to issue 15:
https://github.com/yandex-qatools/juseppe/issues/15

I have launched Juseppe with help of docker on a separate server. On the
same server, I have also set up port mapping using:

docker run -d --name juseppe -v /var/lib/juseppe/plugins:/juseppe/plugins
-p 80:8080 -e JUSEPPE_BASE_URI=http://juseppe.company.net lanwen/juseppe

It seems to be setup correctly because when i upload a new plugin in
"/var/lib/juseppe/plugins"
directory, I can see that plugin becomes available at
http://dockerhost:80/update-center.json

After that within Jenkins server, I configured "UpdateSites Manager plugin". I
have specified ID, URL of the update site and CA for the site.
Still, I don't see plugin within Jenkins server. I cannot find it under Manage
Plugins --> Installed


Thanks,

Hemant
